pkg/apis/core/v1alpha1/tiltrun_types.go

```diff
@@ -159,97 +162,72 @@ type TiltRunStatus struct {
-type BuildStateTerminated struct {
-	// StartTime is when the build began.
+// TargetStateTerminated is a target that finished running, either because it completed successfully or
+// encountered an error.
+type TargetStateTerminated struct {
+	// StartTime is when the target began executing.
 	StartTime metav1.MicroTime `json:"startTime"`
-	// FinishTime is when the build stopped.
+	// FinishTime is when the target stopped executing.
 	FinishTime metav1.MicroTime `json:"finishTime"`
-	// Error is a non-empty string if the build did not complete successfully.
-	Error string `json:"error,omitempty"`
-}
-
-// RuntimeStatus describes the current state of resource execution. Some values might not be used by every RuntimeType.
-type RuntimeStatus string
-
-const (
-	// RuntimeStatusPending indicates that the resource has not yet been started because it is waiting on another
-	// resource.
-	RuntimeStatusPending RuntimeStatus = "pending"
-	// RuntimeStatusRunning indicates that the resource is currently executing.
-	RuntimeStatusRunning RuntimeStatus = "running"
-	// RuntimeStatusSucceeded indicates that the resource ran to completion successfully.
+	// Error is a non-empty string if the target encountered a failure during execution that caused it to stop.
 	//
-	// This is only used for a RuntimeType of "job"; "server" resources never exit during normal operation.
-	RuntimeStatusSucceeded RuntimeStatus = "succeeded"
-	// RuntimeStatusFailed indicates that the resource encountered an error during execution.
-	RuntimeStatusFailed RuntimeStatus = "failed"
-	// RuntimeStatusDisabled indicates that the resource has been requested to not execute at this time.
-	RuntimeStatusDisabled RuntimeStatus = "disabled"
-	// RuntimeStatusUnknown indicates that the status is not currently known; this can occur during initialization.
-	RuntimeStatusUnknown RuntimeStatus = "unknown"
-)
-
-// RuntimeType describes a high-level categorization about the expected behavior for the resource.
-//
-// This can be used in conjunction with RuntimeStatus to fully understand the resource's runtime state.
-type RuntimeType string
-
-const (
-	// RuntimeTypeJob is a resource that is expected to run to completion.
-	RuntimeTypeJob RuntimeType = "job"
-	// RuntimeTypeServer is a resource that runs indefinitely.
-	RuntimeTypeServer RuntimeType = "server"
-)
-
-// RuntimeState describes the current execution state for a resource.
-type RuntimeState struct {
-	// Type is the execution profile for this resource to be used in conjunction with Status.
-	Type RuntimeType `json:"type"`
-	// Status is the current execution status for this resource.
-	Status RuntimeStatus `json:"status"`
-	// Error is a non-empty string describing the failure if Status is "failed".
+	// For targets of type TargetTypeServer, this is always populated, as the target is expected to run indefinitely.
```

Comment:
```suggestion
	// For targets of type TargetTypeServer, this is always populated, as the target is expected to run indefinitely, and thus any termination is an error.
```

Reply:
(I took this suggestion but applied it manually since I mid-making other changes anyway and wanted to line wrap it)